substratevm/src/com.oracle.svm.agent.core/src/com/oracle/svm/agent/core/JNIHandleSetBase.java

```diff
@@ -0,0 +1,176 @@
+            JNIObjectHandle h = Support.jniFunctions().getFindClass().invoke(env, name.get());
+            guarantee(h.notEqual(nullHandle()));
+            return h;
+        }
+    }
+
+    /**
+     * A convenience method to return a global handle to a Java class object.
+     *
+     * If the class is not found, a RuntimeException is thrown.
+     *
+     * @param env JNI environment of the thread running the JVMTI callback.
+     * @param className Java VM type signature of the class.
+     * @return Global JNI handle representing the class object.
+     */
+    protected JNIObjectHandle newClassGlobalRef(JNIEnvironment env, String className) {
+        return newTrackedGlobalRef(env, findClass(env, className));
+    }
+
+    /**
+     * Returns a JNI method ID of a Java method.
+     *
+     * If the method is not found, a RuntimeException is thrown.
+     *
+     * @param env JNI environment of the thread running the JVMTI callback.
+     * @param clazz Handle to the class containing the method.
+     * @param name Name of the method.
+     * @param signature Signature of the method. See the JNI specification for more details.
+     * @param isStatic Specifies whether the method is static or not.
+     * @return JNI method ID of the Java method.
+     */
+    protected static JNIMethodId getMethodId(JNIEnvironment env, JNIObjectHandle clazz, String name, String signature, boolean isStatic) {
+        JNIMethodId id = getMethodIdOptional(env, clazz, name, signature, isStatic);
+        guarantee(id.isNonNull());
+        return id;
+    }
+
+    /**
+     * Returns a JNI method ID of a Java method. If the method is not found, returns a
+     * {@link JNIObjectHandles#nullHandle()}.
+     *
+     * @param env JNI environment of the thread running the JVMTI callback.
+     * @param clazz Handle to the class containing the method.
+     * @param name Name of the method.
+     * @param signature Signature of the method. See the JNI specification for more details.
+     * @param isStatic Specifies whether the method is static or not.
+     * @return JNI method ID of the Java method if found, {@link JNIObjectHandles#nullHandle}
+     *         otherwise.
+     */
+    protected static JNIMethodId getMethodIdOptional(JNIEnvironment env, JNIObjectHandle clazz, String name, String signature, boolean isStatic) {
+        try (CTypeConversion.CCharPointerHolder cname = Support.toCString(name); CTypeConversion.CCharPointerHolder csignature = Support.toCString(signature)) {
+            if (isStatic) {
+                return Support.jniFunctions().getGetStaticMethodID().invoke(env, clazz, cname.get(), csignature.get());
+            } else {
+                return Support.jniFunctions().getGetMethodID().invoke(env, clazz, cname.get(), csignature.get());
+            }
+        }
+    }
+
+    /**
+     * Creates a global JNI handle of the specified local JNI handle.
```

Comment:
Document that this instance remembers the reference and destroys it in `destroy`.

Reply:
Will document this part here too.